features/git-town/configuration/configuration_initial.feature

```diff
@@ -0,0 +1,73 @@
+Feature: Initial configuration
+
+  Background:
+    Given I haven't configured Git Town yet
+
+
+  Scenario Outline: Running Git Town commands while Git Town is unconfigured
+    When I run `<COMMAND>` while allowing errors
+    Then I see
+      """
+      Git Town hasn't been configured for this repository.
+      Please run 'git town config --setup'.
+      Would you like to do that now? y/n
+      """
+
+    Examples:
+    | COMMAND            |
+    | git extract        |
+    | git hack           |
+    | git kill           |
+    | git pr             |
+    | git prune-branches |
+    | git repo           |
+    | git ship           |
+    | git sync           |
+    | git sync-fork      |
+
+
+  Scenario: Enter non-existent main branch
+    When I run `git town config --setup` and enter "nonexistent"
+    Then I see
+      """
+      Please enter the name of the main dev branch (typically 'master' or 'development'):
+
+        Error
+        There is no branch named 'nonexistent'
+
+      """
+    And Git Town is still not configured for this repository
+
+
+  Scenario: Enter valid main branch and non-existent non-feature branch
+    Given I have a branch named "master"
+    When I run `git town config --setup` and enter "master" and "nonexistent"
+    Then I see
+      """
+      Please enter the name of the main dev branch (typically 'master' or 'development'):
+
+      Git Town supports non-feature branches like 'release' or 'production'.
+      These branches cannot be shipped and do not merge master when syncing.
+      Please enter the names of all your non-feature branches as a comma separated list.
+      Example: 'qa, production'
+
+        Error
+        There is no branch named 'nonexistent'
+
+      """
```

Comment:
Just letting you know, I want to change this interface so instead of entering multiple branches as a comma delimited string, just let the user add one branch per line and entering an empty line means the user is done. This can totally be done in a separate PR but may be simpler to just convert this over in this PR.